service_configuration_lib/cached_view.py

```diff
@@ -225,8 +230,9 @@ def process_IN_DELETE(self, event: pyinotify.Event) -> None:
 
     def process_IN_Q_OVERFLOW(self, event: pyinotify.Event) -> None:
         log.warning('Got queue overflow! Recreating watchers.')
-        self.cache.setup()
+        self.cache._needs_reconfigure = True
 
     def process_IN_DELETE_SELF(self, event: pyinotify.Event) -> None:
-        log.warning('Self folder was deleted! Recreating watchers.')
-        self.cache.setup()
+        if event.pathname == os.path.abspath(self.cache._configs_folder):
+            log.warning('Self folder was deleted! Recreating watchers.')
```

Comment:
Can you embed the actual folder name in the message? Makes it a little more useful to uninitiated reader of the warning.

Reply:
Done.